services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
 			if err == nil {
 				t.Fatalf("expected error, got none.")
 			}
-			if err.Error() != tc.expectedErrorMsg {
-				t.Fatalf("Expected different error (expected: %s, got: %s)", tc.expectedErrorMsg, err.Error())
+			var expectedErr api.CreateReleaseResponse
+			if err := prototext.Unmarshal([]byte(tc.expectedErrorMsg), &expectedErr); err != nil {
+				t.Fatalf("failed to unmarshal the expected error object: %v", err)
+			}
```

Comment:
please move this block up, so it is run as early as possible.
This way, if it fails, it's obvious that it's a problem with the test setup and not the test case.